cfe_internal/enterprise/federation/federation.cf

```diff
@@ -206,6 +206,11 @@ bundle agent federation_manage_files
     "feeder_username" data => parsejson('{"feeder_username":"$(cfengine_enterprise_federation:config.transport_user)"}');
     "feeder" data => parsejson('{"feeder": "$(sys.key_digest)"}');
 
+    "DEBUG|DEBUG_config"::
+      "debug" data => parsejson('{"debug":true}');
+    "!(DEBUG|DEBUG_config)"::
```

Comment:
No need for the quotes `"` in the class expression unless you are using variables.

So I would do this:

```
DEBUG|DEBUG_config::
      "debug" data => parsejson('{"debug":true}');
!(DEBUG|DEBUG_config)::
```

OR I would do this:

```
"DEBUG|DEBUG_$(this.bundle)"::
      "debug" data => parsejson('{"debug":true}');
"!(DEBUG|DEBUG_$(this.bundle))"::
```

Also, since this federation policy is in a namespace, I would use the namespace in the guard as well.

```
"DEBUG|DEBUG_$(this.namespace)_$(this.bundle)"::
      "debug" data => parsejson('{"debug":true}');
"!(DEBUG|DEBUG_$(this.namespace)_$(this.bundle))"::
```

or



```
DEBUG|DEBUG_cfengine_enterprise_federation_config::
      "debug" data => parsejson('{"debug":true}');
!(DEBUG|DEBUG_cfengine_enterprise_federation_config)::
```

Reply:
I don't have much of an opinion on the key used for debug in the passed json data.

My first instinct would be to mirror the classes structure to try and make the contexts more familiar, but really, I don't think this matters much at all.

If you wanted to do that, it would be something like this:

```
DEBUG|DEBUG_cfengine_enterprise_federation_config::
      "debug" data => parsejson('{ "classes": { "DEBUG":true}}');
!(DEBUG|DEBUG_cfengine_enterprise_federation_config)::
```
And of course the mustache would need to be updated as well to use `classes.DEBUG` instead of just `debug`.